WebApplication/Backend/Controllers/AppointmentController.cs

```diff
@@ -3,45 +3,80 @@
-
+        private readonly AppointmentService appointmentService;
+        private readonly DateTimeDTO dateTimeDTO;
         public AppointmentController()
         {
-            this.appointmentsService = new AppointmentsService();
+            this.appointmentService = new AppointmentService();
+            this.dateTimeDTO = new DateTimeDTO();
         }
 
         [HttpGet("allAppointmentsByPatientId")]
         public List<Appointment> GellAllAppointmentsByPatientId(String patientId)
         {
-            return appointmentsService.GetAllAppointmentsByPatientId(patientId);
+            return appointmentService.GetAllAppointmentsByPatientId(patientId);
         }
 
+        [HttpGet("physicians")]
+        public List<PhysicianDTO> GetAllPhysitians()
+        {
+            return appointmentService.GetAllPhysicians();
+        }
+        [HttpGet("specializations")]
+        public List<SpecializationDTO> GetAllSpecializations()
+        {
+            return appointmentService.GetAllSpecializations();
+        }
         [HttpGet("allAppointments")]
         public List<Appointment> GellAllAppointments()
         {
-            return appointmentsService.GetAllAppointments();
+            return appointmentService.GetAllAppointments();
         }
 
-
-        [HttpGet("allAppointmentsByPatientIdActive")]
+        //[HttpGet("allAppointmentsByPatientIdActive")]
         public List<Appointment> GellAllAppointmentsByPatientIdActive(String patientId)
         {
-            List<Appointment> apointments = appointmentsService.GetAllAppointmentsByPatientIdActive(patientId);
+            List<Appointment> apointments = appointmentService.GetAllAppointmentsByPatientIdActive(patientId);
             return apointments;
         }
 
-
         [HttpGet("allAppointmentsByPatientIdCanceled")]
         public List<Appointment> GellAllAppointmentsByPatientIdCanceled(String patientId)
         {
-            return appointmentsService.GetAllAppointmentsByPatientIdCanceled(patientId);
+            return appointmentService.GetAllAppointmentsByPatientIdCanceled(patientId);
+        }
+
+        [HttpGet("appointments")]
+        public List<TimeIntervalDTO> GetAllAvailableAppointments(string physicianId, string specializationName, string date)
+        {
+            if (dateTimeDTO.IsPreferredTimeValid(date))
+                return appointmentService.GetAllAvailableAppointments(physicianId, specializationName, date);
+            else
+                return null;
+        }
+
+        [HttpPost("makeAppointment/{physicianId}/{timeIntervalStart}/{date}")]
+        public IActionResult MakeAppointment(string physicianId, DateTime timeIntervalStart, string date)
+        {
+            if (physicianId != null && timeIntervalStart != null && date != null)
+            {
+                if (appointmentService.AddAppointment(new Appointment(new AppointmentDto(physicianId, date, timeIntervalStart))))
```

Comment:
Do you maybe need to check if this appointment could be made first? Is the doctor available at that time, are there any free rooms, is the data valid (not in the past)?

Reply:
Based on the obtained time intervals, this check has already been performed. We only take time intervals for which the doctor is on duty. If I'm not mistaken, another role should assigned the room.